env/fs_on_demand.h

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2024-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+#include <string>
+
+#include "rocksdb/file_system.h"
+
+namespace ROCKSDB_NAMESPACE {
+// A FileSystem that links files to a destination directory from a
+// corresponding source directory on demand. The decision to link
```

Comment:
Just a thought but maybe instead of source and destination, we could use the local/remote terminology across the board